grafast/grafast/src/steps/__inputObject.ts

```diff
@@ -173,6 +173,31 @@ export class __InputObjectStep<
     }
     return !this.inputFields[attrName].step.evalIs(undefined);
   }
+
+  evalKeys(): Array<
+    (keyof TInputType & string) | `${keyof TInputType & number}`
+  > {
+    if (this.inputValues?.kind !== "ObjectValue") {
+      throw new Error("evalKeys must only be called for object types");
+    }
+
+    const keys = new Array<
+      (keyof TInputType & string) | `${keyof TInputType & number}`
+    >();
+    const inputFieldKeys = Object.keys(this.inputFields);
```

Comment:
We can safely assume this will always just be strings, since no other types are permitted.

```suggestion
    const inputFieldKeys = Object.keys(this.inputFields) as string[];
```

This should then mean we don't need the awkward cast in the `keys.push(...)` below.